File: reservation-service/src/main/java/ma/esnet/conferenceservice/enities/Reservation.java
package ma.esnet.conferenceservice.enities;

import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import ma.esnet.conferenceservice.model.Ressource;


import java.util.Date;

@Entity
@Data @AllArgsConstructor @NoArgsConstructor @Builder
public class Reservation {
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String nom;
    private String context;
    private Date date;
    private Long ressourceId;
    private Double duree;
    @ManyToOne
    @JsonProperty(access = JsonProperty.Access.WRITE_ONLY)
    private Personne personne;
    @Transient
    private Ressource ressource;

}

File: AngularClient/src/app/model/Reservation.ts
export class Reservation {
  id: number;
  nom: string;
  context: string;
  date: string;
  duree: number;
  ressourceId: number;
  personne: {
    id: number;
    nom: string;
    email: string;
    fonction: string;
  };
  ressource: {
    id: number;
    nom: string;
    ressourceType: string;
  };

  constructor() {
    this.id = 0;
    this.nom = '';
    this.context = '';
    this.date = '';
    this.duree = 0;
    this.ressourceId = 0;
    this.personne = { id: 0, nom: '', email: '', fonction: '' };
    this.ressource = { id: 0, nom: '', ressourceType: '' };
  }
}
